pkg/registry/generic/etcd/etcd.go

```diff
@@ -441,6 +441,9 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	// delete immediately, or no graceful deletion supported
 	out := e.NewFunc()
 	if err := e.Storage.Delete(ctx, key, out); err != nil {
+		if storage.IsNotFound(err) {
```

Comment:
doesn't returning `errDeleteNow` in the simple update func above mean etcd isn't updated? I think that only in this case is this handler actually the one causing the delete:
```
lastGraceful = *options.GracePeriodSeconds
return existing, nil
```